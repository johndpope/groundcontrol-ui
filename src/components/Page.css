.Page {
  margin: 2em 2em 1em;
}

.PageFullWidth {
  margin: 2em 0 1em;
}

.PageFullWidth .ui.header {
  margin: 0 1em;
}

@media only screen and (max-width: 767px) {
  .Page {
    margin-left: 0;
    margin-right: 0;
  }
}

.Page > .ui.header {
  margin-bottom: 1em;
}
