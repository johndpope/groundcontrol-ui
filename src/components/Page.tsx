
// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from "react";
import {
  Container,
  Header,
  Icon,
  SemanticICONS,
} from "semantic-ui-react";

import "./Page.css";

export interface IProps {
  className?: string;
  icon: SemanticICONS;
  header: string;
  subheader: string;
  children: React.ReactNode;
}

const Page = ({
  className,
  icon,
  header,
  subheader,
  children,
}: IProps) => (
  <div className={`Page ${className || ""}`}>
    <Header as="h1">
      <Icon name={icon} />
      <Header.Content>
        {header}
        <Header.Subheader>{subheader}</Header.Subheader>
      </Header.Content>
    </Header>
    <Container fluid={true}>
      {children}
    </Container>
  </div>
);

export default Page;
