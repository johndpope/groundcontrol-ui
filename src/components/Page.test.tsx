// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { shallow } from "enzyme";
import React from "react";

import Page from "./Page";

describe("<Page />", () => {

  it("renders correctly", () => {
    const wrapper = shallow(
      <Page
        header="header"
        subheader="subheader"
        icon="rocket"
      >
        <p>Hello, World!</p>
      </Page>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly with a CSS class", () => {
    const wrapper = shallow(
      <Page
        header="header"
        subheader="subheader"
        icon="rocket"
        className="className"
      >
        <p>Hello, World!</p>
      </Page>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly without a container", () => {
    const wrapper = shallow(
      <Page
        header="header"
        subheader="subheader"
        icon="rocket"
        container={false}
      >
        <p>Hello, World!</p>
      </Page>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders children when passed in", () => {
    const wrapper = shallow(
      <Page
        header="header"
        subheader="subheader"
        icon="rocket"
      >
        <div className="unique" />
      </Page>,
    );
    expect(wrapper.contains(<div className="unique" />)).toEqual(true);
  });

});
