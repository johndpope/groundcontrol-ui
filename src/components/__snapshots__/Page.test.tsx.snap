// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders correctly 1`] = `
<div
  className="Page  "
>
  <Header
    as="h1"
  >
    <Icon
      as="i"
      name="rocket"
    />
    <HeaderContent>
      header
      <HeaderSubheader>
        subheader
      </HeaderSubheader>
    </HeaderContent>
  </Header>
  <Container
    fluid={true}
  >
    <p>
      Hello, World!
    </p>
  </Container>
</div>
`;

exports[`<Page /> renders correctly when fullWidth is set to true 1`] = `
<div
  className="Page PageFullWidth "
>
  <Header
    as="h1"
  >
    <Icon
      as="i"
      name="rocket"
    />
    <HeaderContent>
      header
      <HeaderSubheader>
        subheader
      </HeaderSubheader>
    </HeaderContent>
  </Header>
  <Container
    fluid={true}
  >
    <p>
      Hello, World!
    </p>
  </Container>
</div>
`;

exports[`<Page /> renders correctly with a CSS class 1`] = `
<div
  className="Page  className"
>
  <Header
    as="h1"
  >
    <Icon
      as="i"
      name="rocket"
    />
    <HeaderContent>
      header
      <HeaderSubheader>
        subheader
      </HeaderSubheader>
    </HeaderContent>
  </Header>
  <Container
    fluid={true}
  >
    <p>
      Hello, World!
    </p>
  </Container>
</div>
`;
